### ISSUE TITLE
🤖 [AUTO] Implement Strategy: fama_french_long_short_equity

### ISSUE THREAD

**Issue by kpakpooluwaseyi** [kpakpooluwaseyi/Agentic-Algo-Strategies#573]
## 🤖 Auto-Generated Strategy Request

**Source:** Generative AI for Trading 2025.pdf (strategy 2)
**Extracted:** 2025-12-27T06:48:34.919956

---

1.  **Strategy Name**: fama_french_long_short_equity
2.  **Strategy Type**: Factor-based, Long-Short, Predictive
3.  **Timeframe**: Daily
4.  **Instruments**: Individual stocks (implied US equities)

5.  **Entry Rules**:
    *   **Prediction Model**: For each stock, train a linear regression model to predict its next day's return ($R_{stock,t+1}$) using the current day's Fama-French three factors ($Mkt-RF_t$, $SMB_t$, $HML_t$) as independent variables. The model is:
        $R_{stock,t+1} = \alpha + \beta_1 \cdot Mkt-RF_t + \beta_2 \cdot SMB_t + \beta_3 \cdot HML_t + \epsilon$
    *   **Selection**: On each trading day,
        *   **Long Entry**: Identify the 50 stocks with the highest predicted next day's returns. Buy these stocks.
        *   **Short Entry**: Identify the 50 stocks with the lowest predicted next day's returns. Short these stocks.

6.  **Exit Rules**:
    *   **Time-based Exit**: Positions are held for a specific period.
        *   Originally proposed: 30 days.
        *   Simplified for backtest example in the text: 1 day (positions are closed at the end of the day or before the next trading decision).
    *   **Take Profit**: Not explicitly defined.
    *   **Stop Loss**: Not explicitly defined.

7.  **Indicators Required**:
    *   Fama-French Three Factors: Mkt-RF (Market Excess Return), SMB (Small Minus Big), HML (High Minus Low).
    *   Historical daily stock prices (to compute daily returns).
    *   A linear regression model (e.g., from scikit-learn in Python).

8.  **Risk Management**:
    *   **Position Sizing**: Equal number of stocks (50 long, 50 short). Implied dollar-neutral if positions are sized equally in dollar terms.
    *   **Maximum Risk Per Trade**: Not explicitly defined.
    *   **Risk-Reward Ratio**: Not explicitly defined.
    *   **Overall Portfolio**: Designed to be dollar-neutral.

9.  **Additional Notes**:
    *   **Model Training**: Regression models are trained using a historical training dataset (e.g., the first half of available data) and then applied to an out-of-sample test dataset for predictions.
    *   **Data Handling**: Rows with NaN values in relevant columns (returns or factors) are removed (e.g., via `dropna()`) before fitting the linear regression models. A minimum data threshold (e.g., 100 data points) is applied for training a stock's model.
    *   **Sharpe Ratio Calculation**: For a self-financing, dollar-neutral portfolio, the risk-free rate is typically *not* subtracted from raw returns when calculating the Sharpe ratio, assuming margin interest/credit is not factored into the P&L calculation. The backtest example sets the risk-free rate to 0.

---

## Implementation Instructions

Please implement this strategy using the `backtesting.py` framework with the following requirements:

1. **File Location:** `strategies/fama_french_long_short_equity.py`
2. **Data Path:** Use `data/BTC-USD-15m.csv` for backtesting
3. **Indicators:** Use `talib` or `pandas_ta` (NOT backtesting.py's built-in indicators)
4. **Template:** Follow existing strategies in the `strategies/` folder

### Required Components:
- Strategy class inheriting from `Strategy`
- Proper indicator initialization using `self.I()`
- Entry/exit logic in `next()` method
- Default parameters as class attributes
- Risk management (stop loss, take profit)

### Data Format:
```
datetime, open, high, low, close, volume
2023-01-01 00:00:00, 16531.83, 16532.69, 16509.11, 16510.82, 231.05
```

### Output:
- Run backtest with default parameters
- Print stats and create HTML plot
- Save results to `results/temp_result.json`
